sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1912,6 +1913,26 @@ class Analyzer(
           nondeterToAttr.get(e).map(_.toAttribute).getOrElse(e)
         }.copy(child = newChild)
 
+      case j: Join if j.condition.isDefined && !j.condition.get.deterministic =>
+        j match {
+          // We can push down non-deterministic joining keys.
+          // We can't push down non-deterministic conditions.
+          case ExtractEquiJoinKeys(_, leftKeys, rightKeys, conditions, _, _)
```

Comment:
We do not support non-deterministic join condition. Thus, our current execution orders in the join implementation might not behave correctly.

If we really need to support it, we have to check what is the right behavior in the traditional DB system.